codecs/jxl/Makefile

```
@@ -51,18 +51,8 @@ $(CODEC_OUT): $(CODEC_DIR)/CMakeLists.txt
 $(CODEC_DIR)/CMakeLists.txt: $(CODEC_DIR)
 
 $(CODEC_DIR):
-	# The JXL repository doesn’t have version tags or anything yet,
-	# so we have to pin to a specific commit for now. This implies we
-	# can’t use --recursive, as we will change commit after checkout (it
-	# seems you can’t clone a specific commit directly), and it also means
-	# we can’t use --depth 1 because we want to change commits.
-	# The JXL code base also relies on submodules so we can’t just download
-	# a .tar.gz from GitLab.
 	mkdir -p $@
-	git clone $(CODEC_URL) $@
-	cd $@ && \
-	git reset --hard $(CODEC_VERSION) && \
-	git submodule update --init --recursive
+	git clone $(CODEC_URL) --recursive --depth 1 --branch $(CODEC_VERSION) $@
```

Ha, on another branch I just recently made a different change to allow parallel submodules even without `clone`, but this simplifies it a lot. Just one suggestion - let's parallelise it:

```suggestion
	git clone $(CODEC_URL) --recursive -j`nproc` --depth 1 --branch $(CODEC_VERSION) $@
```